api_v3/lib/types/entry/KalturaBaseEntry.php

```diff
@@ -541,24 +550,53 @@ public function doFromObject($sourceObject, KalturaDetachedResponseProfile $resp
 				$this->creatorId = $sourceObject->getCreatorPuserId();
 		}
 		$requestLanguage = kCurrentContext::getLanguage();
-		if($requestLanguage)
+		
+		$this->injectCorrectLanguageValues($sourceObject, $requestLanguage);
+	}
+	
+	protected function injectCorrectLanguageValues($sourceObject, $requestLanguage = null)
+	{
+		$multiLanguageMap = json_decode($sourceObject->getMultiLanguageMapping(), true);
+		if ($requestLanguage == multiLingualUtils::MULTI)
 		{
-			$this->injectCorrectLanguageValues($sourceObject, $requestLanguage);
+			
+			$this->setMultiLanguageStringInField($sourceObject, $multiLanguageMap);
+		}
+		else
+		{
+			$this->setRequestedLanguageStringInField($multiLanguageMap, $sourceObject, $requestLanguage);
 		}
 	}
 	
-	protected function injectCorrectLanguageValues($sourceObject, $requestLanguage)
+	protected function setMultiLanguageStringInField($sourceObject, $multiLanguageMap)
 	{
-		$multiLangMapping = json_decode($sourceObject->getMultiLanguageMapping(), true);
-		$name = $sourceObject->extractLanguageValue($multiLangMapping, self::NAME, $requestLanguage);
-		$description = $sourceObject->extractLanguageValue($multiLangMapping, self::DESCRIPTION, $requestLanguage);
-		$tags = $sourceObject->extractLanguageValue($multiLangMapping, self::TAGS, $requestLanguage);
-		$tags = ktagword::updateTags($this->tags, $tags , false );
-		if($multiLangMapping)
+		$defaultLanguage = $sourceObject->getObjectDefaultLanguage();
+		$supportedFields = $sourceObject->getMultiLingualSupportedFields();
+		foreach ($supportedFields as $fieldName)
 		{
-			$this->name = $name ? $name : $this->name;
-			$this->description = $description ? $description : $this->description;
-			$this->tags = $tags ? $tags : $this->tags;
+			if ($this->$fieldName)
+			{
+				$multiLanguageMap[$fieldName][$defaultLanguage] = $sourceObject->getDefaultFieldValue($fieldName);
+			}
+			$this->$fieldName = KalturaKeyValueArray::fromKeyValueArray($multiLanguageMap[$fieldName]);
+		}
+	}
+	
+	protected function setRequestedLanguageStringInField($multiLangMapping, $sourceObject, $requestLanguage = null)
+	{
+		$language = $requestLanguage ? $requestLanguage : $this->entryDefaultLanguage;
+		$supportedFields = $sourceObject->getMultiLingualSupportedFields();
+		$supportedFieldsInRequestedLang = array();
+		foreach ($supportedFields as $fieldName)
+		{
+			$supportedFieldsInRequestedLang[$fieldName] = multiLingualUtils::extractLanguageValue($multiLangMapping, $fieldName, $language);
+			if ($fieldName == 'tags')
+			{
+				$supportedFieldsInRequestedLang[$fieldName] = ktagword::updateTags($this->tags, $supportedFieldsInRequestedLang[$fieldName], false);
```

Comment:
why we have this here again? it looks redundant
if you will handle it differently the function can be general and move d from this class as well